src/core/server/plugins/legacy_config.ts

```diff
@@ -28,8 +30,19 @@ const createGlobalConfig = ({
   path: PathConfigType;
   savedObjects: SavedObjectsConfigType;
 }): SharedGlobalConfig => {
+  const resolvedElasticsearchConfig = new ElasticsearchConfig(elasticsearch);
+  const caFingerprints =
+    resolvedElasticsearchConfig.ssl.certificateAuthorities?.map(
+      (ca) => new X509Certificate(Buffer.from(ca)).fingerprint256
+    ) ?? [];
+
   return deepFreeze({
-    elasticsearch: pick(elasticsearch, SharedGlobalConfigKeys.elasticsearch),
+    elasticsearch: {
+      ...pick(elasticsearch, SharedGlobalConfigKeys.elasticsearch),
+      ssl: {
+        certificateAuthorityFingerprints: caFingerprints,
+      },
+    },
```

Comment:
I wonder if we really want to use the deprecated 'legacy config' API for this, or if we should make a proper API exposed from the ES service instead? 

My thinking is
- We want to remove this legacy config at some point, and we've been trying to only remove properties from them, until complete removal
- This config was supposed to only expose 'raw' values from the config, and here, we need to process the certs by instantiating an `ElasticsearchConfig` to convert them.
- If we are to expose these fingerprints to plugins, introducing an official API from the ES service seems cleaner

The only argument against it imho would be if this is meant to be a temporary workaround that is supposed to be removed in the short term.

What do the other @elastic/kibana-core members think?

Reply:
> The only argument against it imho would be if this is meant to be a temporary workaround that is supposed to be removed in the short term.

This won't be temporary. I thought about this as well but this feels out of scope for what we want to get in for 8.0 at this point?

Reply:
I agree that the timing isn't necessarily on our side here. @lukeelmers @rudolf wdyt?

Reply:
We already have `elasticsearch.legacy.config$`, and I think this PR would already be exposing the fingerprints through that API. So moving `certificateAuthorities?.map((ca) => ca.fingerprint256)` into the ElasticsearchService seems necessary already.

If there's no way fleet can fit moving this into a dedicated API before 8.0 I think we should create an issue that the core team can pick up in our next sprint.

Reply:
> If there's no way fleet can fit moving this into a dedicated API before 8.0 I think we should create an issue that the core team can pick up in our next sprint.

Agree on this -- I don't think this should live exclusively in the legacy config unless it is only meant to be temporary. ES service feels like the right home for it longer term.

Reply:
Since there are several considerations that need to be made here (naming, contents, etc.) before adding a stable API, I am going to move forward with extending the existing interface and opened a new issue for the cleanup to create a stable API. I believe there are other config values that are also needed throughout Kibana today.